### PR TITLE
Watch changes on storage and change state (closes #27)

### LINKED ISSUE #27: New feature -> watch localstorage??
Hi, I was using your module (thanks by the way :-) ) and I encountered a problem. I was in a situation where one component was updating localstorage and another one was 'watching' it. And if localstorage was updated, the watching component needed to rerender with the new data.
But the way the hook is written, it doesn't 'watch' localstorage.

So in my project, I modified your hook and added the following effect:
```
  useEffect(() => {
    const checkValue = localStorage.getItem(key)
    if (checkValue  !== value) {
      setValue(checkValue )
    }
  });
```
This way, the state is updated when localstorage changes, and each component using the hook rerenders with the new data. For me it works perfectly, but perhaps you might see a downside (besides added complexity and extra calls to localStorage).
It could also be hidden behind an option.

So anyhow, do want you want with it :-) but I just wanted to let you know what I've done to solve my problem :-)

### PR DESCRIPTION
**Pull request by VitorLuizC**
I added a listener on storage that runs when modifying storage in another window or tab and an effect that runs on all cycles. Both change state if storage changes occur.

closes #27

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1,17 +1,38 @@
...
+      if (event.key === key && event.newValue !== value) {
+        setValue(event.newValue || initialValue);
+      }
+    },
+    [value]
+  );
+
+  useEffect(() => {
+    window.addEventListener('storage', handleStorage);
+    return () => window.removeEventListener('storage', handleStorage);
```
Comment (Svish) [inline src/index.ts]: If `value` changes, causing `handleStorage` callback to change, will `removeEventListener` still work here then? 🤔

Comment (VitorLuizC) [reply src/index.ts]: It will remove and append again in the next cycle.

**Comment by dance2die** [comment]
Thank you @VitorLuizC for the PR (turn off notifications w/ checking 😓) and @Svish for the review.  
I will check test the PR soon I get the chance (weekend latest).

**Comment by dance2die** [comment]
Thank you again, @VitorLuizC.
The change is implemented in [v3.4.1](https://www.npmjs.com/package/react-use-localstorage/v/3.4.1)

**Comment by dance2die** [comment]
@all-contributors please add @Svish for review

**Comment by Svish** [comment]
@dance2die Really not sure my question counts as a contribution 😛

**Comment by dance2die** [comment]
You were able to come up with the question because you read the code, spending your own time @Svish.  
No matter how small a contribution is, it is still to be recognized/credited 😄

**Pull request merged**